scripts: check_init_priorities: fix device function name display

fabiobaltieri opened the pull request:
Device init pointers have been moved off struct init_entry and into struct device in 766bfe7b2e2, but check_init_priorities.py have not been modified to check the new pointer so it's been showing NULL for all devices since.

Fix it by searching down the right pointer for device init entries.

fabiobaltieri committed:
scripts: check_init_priorities: fix device function name display

Device init pointers have been moved off struct init_entry and into
struct device in 766bfe7b2e2, but check_init_priorities.py have not been
modified to check the new pointer so it's been showing NULL for all
devices since.

Fix it by searching down the right pointer for device init entries.

Signed-off-by: Fabio Baltieri <fabiobaltieri@google.com>

kartben reviewed (dismissed):
Ah! I knew something was definitely off last time I used it and wanted to ask you :)

pdgendt commented on scripts/build/check_init_priorities.py:
```diff
@@ -119,6 +123,7 @@ def _load_objects(self):
                     sym.entry.st_info.type in ["STT_OBJECT", "STT_FUNC"]):
                     self._objects[sym.entry.st_value] = (
                             sym.name, sym.entry.st_size, sym.entry.st_shndx)
+                    self._object_addr[sym.name] = sym.entry.st_value;
```
Need to drop the semicolon

fabiobaltieri replied:
yup compliance found it too

pdgendt reviewed (dismissed):
I guess it's not trivial to catch these kind of issues in the future?

fabiobaltieri committed:
test: check_init_priorities: add a test for initlevels

Add a test for check_init_priorities --initlevels in addition to the
normal error output, this validates that the function names are printed
correctly.

The current case covers sys_init, device with no init functions and
device with init function so it should be all cases.

Signed-off-by: Fabio Baltieri <fabiobaltieri@google.com>

fabiobaltieri commented:
Sorry looks like I forgot how to write C after forgetting how to write Python, should be alright now. :-)
